### PR TITLE
Add tags to create_job, and add an update_job function to the sdk

### PR DESCRIPTION
**Pull request by ari-nz**
Add more REST api functionality to the python SDK

Issue #456

### CODE REVIEW

File: databricks_cli/sdk/service.py [hunk 1]
```diff
@@ -130,6 +132,18 @@ def reset_job(self, job_id, new_settings, headers=None, version=None):
                 raise TypeError('Expected databricks.JobSettings() or dict for field new_settings')
         return self.client.perform_query('POST', '/jobs/reset', data=_data, headers=headers, version=version)
 
+    def update_job(self, job_id, new_settings, fields_to_remove=None, headers=None, version=None):
```
Comment (nfx) [inline databricks_cli/sdk/service.py:135]: @ari-nz please also cover the corner cases with additional unit tests.

**Review by nfx — CHANGES_REQUESTED** [review]

**Comment by ari-nz** [comment]
@nfx  - I've added a few more tests. Let me know if there's anything else specific that you need.

**Comment by ari-nz** [comment]
@nfx I've rebased this fork back onto the latest version of main. It looks like the `create_job` api has been updated, but not the `update_job`

Requesting this PR to be merged

**Comment by ari-nz** [comment]
Tagging @pietern too as it looks like you've been actively lately :)

**Comment by pietern** [comment]
@ari-nz Thanks for the ping. We've picked up generating `service.py` from an internal definition of the API as of a couple of months ago, which is why the updated version of `create_job` appeared. I'll perform a regeneration and include `update_job` and pull in your tests.
